Climber inital implementation

aaron97neu opened the pull request:
Should be ready for review. Needs further work once physical climber is built

- [x] Code builds successfully
- [x] Code simulates successfully

NoahRen25 committed:
subsystem necesito mucho trabajo

NoahRen25 committed:
command and subsystem done for now for climber

KomeiOkamoto12390 committed:
Path update

jruben4 committed:
buggy interpolation

NoahRen25 committed:
hasNote for new bot

KomeiOkamoto12390 committed:
WIP

dlaufen committed:
ConstantVelocityTest, imports cleanups

dlaufen committed:
Tree interpolate working example

jruben4 committed:
add getter

NoahRen25 committed:
Merge remote-tracking branch 'origin/SpeakerShooting2' into NewBotSwap

jruben4 committed:
add getDistanceToPose to drivertrain

NoahRen25 committed:
merge

NoahRen25 committed:
Merge branch 'hola' into NewBotSwap

NoahRen25 committed:
shooter angle pos calc in shooterservo subsystem

KomeiOkamoto12390 committed:
WIP

KomeiOkamoto12390 committed:
Merge branch 'master' into autoPath

NoahRen25 committed:
wip

NoahRen25 committed:
wip v2 (done besides stop at same pos, trying to put in ss, logic WIP)

NoahRen25 committed:
need to add some nt

NoahRen25 committed:
nt and corrections bc im dumb

NoahRen25 committed:
actually ready

jruben4 committed:
a few changes on servo distance estimatation workflow

jruben4 committed:
shooter prints fix

beastsoftware2202 committed:
"merged" lol

beastsoftware2202 committed:
v2

beastsoftware2202 committed:
misc changes

jruben4 committed:
shooter servo zeroing untested

jruben4 committed:
add prints

beastsoftware2202 committed:
wip to merge

beastsoftware2202 committed:
Merge remote-tracking branch 'origin/ServoZero' into NewBotTest

beastsoftware2202 committed:
wip

KomeiOkamoto12390 committed:
WIP

beastsoftware2202 committed:
all moving (wip tuning)

aaron97neu committed:
Additl comments w/ gavin on subsystem

aaron97neu committed:
More comments to commands and subsystem

GavinArchon committed:
Added button bindings

climb up and retract climber

aaron97neu commented on src/main/java/frc/robot/subsystems/Climber/Climber.java:
```diff
@@ -0,0 +1,155 @@
+        .burnFlash();
+  }
+
+  /**
+   * Extend arm to position given.
+   *
+   * @param pos Desired position in cm from fully retracted position
+   */
+  public void setArmHeight(double pos) {
```
@GavinArchon and I figured out that this style of comment (JavaDoc maybe?) can be read by VSCode and is show when calls to this method are hovered over. Extremely helpful and might try to make this a habit for the robot

aaron97neu replied:
Syntax ref: https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#format

KomeiOkamoto12390 replied:
You guys overided my javadoc in faceTotag lol

GavinArchon committed:
Started using javaDocs

Used for organization and more readable code.

aaron97neu commented:
Does not simulate due to subsystem config issues atm

beastsoftware2202 committed:
WIP

KomeiOkamoto12390 committed:
AutoPathWIP without subsystem/commands changes

KomeiOkamoto12390 committed:
Command/Subsystem Changes - LOOK AT INTAKE PLEASE

ShooterSequence should be only used for Alpha so its fine but check intake commands and subsystem

KomeiOkamoto12390 committed:
Name change

KomeiOkamoto12390 committed:
Missed some

KomeiOkamoto12390 committed:
WIP done for 3/9

jruben4 committed:
moved auto angle adjustment out into a command per DL

dlaufen committed:
naming conventions shooter. Angle tracker move to cmd

dlaufen committed:
Merge branch 'NewBotTest' of https://github.com/2202Programming/FRC-2024 into NewBotTest

dlaufen committed:
DrJ and I both moved autoAngle code.  Fixed some bugs.

dlaufen committed:
Removed Transfer from ShooterServo SS

dlaufen committed:
Removed redundant zeroEncoder API, Extension name convention.

dlaufen committed:
Shorter naming convention

dlaufen committed:
Test<cmd> naming convention

dlaufen committed:
optional<DS.alliance>  cleanup

beastsoftware2202 committed:
wip  laptop issues

beastsoftware2202 committed:
wip

beastsoftware2202 committed:
wip

KomeiOkamoto12390 committed:
WIP 3 auto tested

KomeiOkamoto12390 committed:
Changes for speaker shoot

beastsoftware2202 committed:
done for now

beastsoftware2202 committed:
Merge remote-tracking branch 'origin/autoPath' into NewBotTest

beastsoftware2202 committed:
Merge remote-tracking branch 'origin/autoPath' into NewBotTest

beastsoftware2202 committed:
WIP

dlaufen committed:
ABS encoder 180 fix, FL flip drive mtr back to true.

NoahRen25 committed:
one motor

KomeiOkamoto12390 commented:
> Does not simulate due to subsystem config issues atm

It did simulate successfully. Just adding subsystem for simulation on betabot.

NoahRen25 committed:
nt and configs

beastsoftware2202 committed:
Gears out calibration, rotate still weird

EdricZhou committed:
Finished some TODO's for Climber

Implemented addRequirements in Climb.java and ClimbVel.java,  added a getCurrent method in Climber.java

KomeiOkamoto12390 committed:
WIP

NoahRen25 committed:
added command

beastsoftware2202 committed:
try again at rotation fix.

beastsoftware2202 committed:
Merge branch 'NewBotTest' of https://github.com/2202Programming/FRC-2024 into NewBotTest

NoahRen25 committed:
Merge branch 'NewBotTest' into climberDraft

KomeiOkamoto12390 committed:
Drivetrain works

NoahRen25 committed:
const

NoahRen25 committed:
Merge remote-tracking branch 'origin/NewBotTest' into climberDraft

KomeiOkamoto12390 committed:
e

NoahRen25 committed:
e

KomeiOkamoto12390 committed:
climber works (vel)

NoahRen25 committed:
climber calibrate stuff

NoahRen25 committed:
testing
